src/ai/.x/templates/openai-chat-streaming-with-functions-js/Main.js

```diff
@@ -14,28 +19,52 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+// NOTE: Never deploy your key in client-side environments like browsers or mobile apps
+//  SEE: https://help.openai.com/en/articles/5112595-best-practices-for-api-key-safety
+
 async function main() {
 
-  const openAIAPIKey = process.env["AZURE_OPENAI_API_KEY"] || "<#= AZURE_OPENAI_API_KEY #>";
-  const openAIEndpoint = process.env["AZURE_OPENAI_ENDPOINT"] || "<#= AZURE_OPENAI_ENDPOINT #>";
-  const openAIChatDeploymentName = process.env["AZURE_OPENAI_CHAT_DEPLOYMENT"] || "<#= AZURE_OPENAI_CHAT_DEPLOYMENT #>" ;
-  const openAISystemPrompt = process.env["AZURE_OPENAI_SYSTEM_PROMPT"] || "<#= AZURE_OPENAI_SYSTEM_PROMPT #>" ;
+  // What's the system prompt
+  const openAISystemPrompt = process.env["AZURE_OPENAI_SYSTEM_PROMPT"] || "You are a helpful AI assistant.";
+
+  // Connection info
+  const azureOpenAIAPIKey = process.env["AZURE_OPENAI_API_KEY"] || "<#= AZURE_OPENAI_API_KEY #>";
+  const azureOpenAIAPIVersion = process.env["AZURE_OPENAI_API_VERSION"] || "<#= AZURE_OPENAI_API_VERSION #>";
+  const azureOpenAIEndpoint = process.env["AZURE_OPENAI_ENDPOINT"] || "<#= AZURE_OPENAI_ENDPOINT #>";
+  const azureOpenAIChatDeploymentName = process.env["AZURE_OPENAI_CHAT_DEPLOYMENT"] || "<#= AZURE_OPENAI_CHAT_DEPLOYMENT #>";
+  const openAIAPIKey = process.env["OPENAI_API_KEY"] || "<#= OPENAI_API_KEY #>";
+  const openAIOrganization = process.env["OPENAI_ORG_ID"] || null;
+  const openAIModelName = process.env["OPENAI_MODEL_NAME"] || "<#= OPENAI_MODEL_NAME #>";
+
+  // Create the OpenAI client
+  const useAzure = azureOpenAIEndpoint?.startsWith("https://");
+  const openai = useAzure
+    ? CreateOpenAI.fromAzureOpenAIKeyAndDeployment(azureOpenAIAPIKey, azureOpenAIEndpoint, azureOpenAIAPIVersion, azureOpenAIChatDeploymentName)
+    : CreateOpenAI.fromOpenAIKey(openAIAPIKey, openAIOrganization);
 
-  const chat = new <#= ClassName #>(openAIEndpoint, openAIAPIKey, openAIChatDeploymentName, openAISystemPrompt, factory);
+  // Create the streaming chat completions helper
+  const chat = streamingChatCompletions = useAzure
+    ? new <#= ClassName #>(azureOpenAIChatDeploymentName, openAISystemPrompt, factory, openai, 20)
+    : new <#= ClassName #>(openAIModelName, openAISystemPrompt, factory, openai);
 
+  // Loop until the user types 'exit'
   while (true) {
 
+    // Get user input
     const input = await rl.question('User: ');
     if (input === 'exit' || input === '') break;
 
-    let response = await chat.getChatCompletions(input, (content) => {
-      console.log(`assistant-streaming: ${content}`);
+    // Get the response
+    process.stdout.write('\nAssistant: ');
+    await chat.getResponse(input, (content) => {
+      process.stdout.write(content);
     });
 
-    console.log(`\nAssistant: ${response}\n`);
+    process.stdout.write('\n\n');
   }
 
   console.log('Bye!');
```

Comment:
Some of the prints were changes to process.stdout.write() should this also be changed?

Reply:
not as used...